packages/grafana-ui/src/components/NewTable/NewTable.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { FC, useMemo, useState } from 'react';
+import { GrafanaTheme } from '@grafana/data';
+import { css, cx } from 'emotion';
+import { stylesFactory, useTheme } from '../../themes';
+import { Icon } from '../Icon/Icon';
+
+interface TableColumn {
+  name: string;
+  render?: (data: any, i: number, row: any[]) => React.ReactNode;
+  sortable?: boolean;
+  sortKey?: (data: string) => string;
+}
+
+export interface Props {
+  headers: TableColumn[];
+  rows: any[][];
+  tableClass?: string;
+}
+
+enum SortDirection {
+  Ascending = 'ascending',
+  Descending = 'descending',
+  None = 'none',
+}
+
+export const NewTable: FC<Props> = (props: Props) => {
```

Comment:
this is still heavily a WIP and will be split into a separate PR

Reply:
Think finding a way to re-use the core Table component would be preferable, what was the main reason for not using it?

Reply:
Main reason was that the existing Table component uses a `DataFrame` as the interface for displaying data, which makes sense for datasource responses but seemed like kind of a strange fit for simpler data.

Reply:
But creating a DataFrame with say toDataFrame is pretty simple. This feels pretty complex as well. Maybe a wrapper around the current Table to make it easier to use for non DataFrame could be an option?